packages/compiler-core/src/parse.ts

```diff
@@ -236,13 +241,8 @@ function parseChildren(
         } else {
           node.content = node.content.replace(/[\t\r\n\f ]+/g, ' ')
         }
-      }
-      // also remove comment nodes in prod by default
-      if (
-        !__DEV__ &&
-        node.type === NodeTypes.COMMENT &&
-        !context.options.comments
-      ) {
+      } else if (node.type === NodeTypes.COMMENT && !context.options.comments) {
```

Comment:
This should not be an `else-if`.
This means that comments will not be stripped out of `<pre>` elements which was one of the goals of this commit:

https://github.com/vuejs/vue-next/commit/f4119249f2d3f394469028ad9664f61830540ff9
> consistently remove comment nodes for pre tags in production